### PR TITLE
Unshaped component

### PR DESCRIPTION
**Pull request by keckler**
## Description

This addresses #591 by allowing for `UnshapedComponent`s to thermally expand. Tests have been added to ensure that the thermal expansion is consistent with the changes in density of the material with temperature. Other existing tests have been modified to prove that the new functionality does not break any existing features.

Also, a bug in the `getBoundingCircleOuterDiameter` method was fixed.

---

## Checklist

- [X] The code is understandable and maintainable to people beyond the author.
- [X] Tests have been added/updated to verify that the new or changed code works.
- [X] There is no commented out code in this PR.
- [X] The commit message follows [good practices](https://terrapower.github.io/armi/developer/tooling.html).
- [X] All docstrings are still up-to-date with these changes.

If user exposed functionality was added/changed:

- [ ] Documentation added/updated in the `doc` folder.
- [ ] New or updated dependencies have been added to `setup.py`.

### CODE REVIEW

File: armi/reactor/tests/test_components.py
Comment (john-science) [inline armi/reactor/tests/test_components.py:236]: Would this perhaps be more clear if it said `self.assertGreater(thermalExpansionFactor, 1)`?

**Review by john-science — APPROVED** [review]
I appreciate the lift.

See my one comment on the unit test.  And change it if you like.

Otherwise, let me wait until after work hours today to merge this?  Thanks.

File: armi/reactor/tests/test_components.py
Comment (keckler) [reply armi/reactor/tests/test_components.py:236]: Sure, I could see that. I'll change.

**Comment by keckler** [comment]
No problem on any delay. Thanks for taking a look!

File: armi/reactor/components/__init__.py
Comment (albeanth) [inline armi/reactor/components/__init__.py:124]: I'd consider even taking this a step further:
1. the sum of the areal fractions of unshaped components need to be equal with the total area of the block. (but maybe there are other places this could/should go)
2. expanding on the neutronics argument that this type is useful for homogenization-like studies when particular shapes/locations of components is not important (since they end up getting smeared together anyway) but rather knowing the area (and ultimately the volume) of a material within a block and how it can change with temperature.

Just a thought.

Comment (john-science) [reply armi/reactor/components/__init__.py:124]: Sorry, I've already merged the PR.  But @keckler , do you have any thoughts here?

**Pull request merged**
